projects/interfaces/main.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+)
+
+type OurByteBuffer struct {
+	myBuff []byte
+}
+
+// Is there a maximum about of data an OurByteBuffer can store?
+// There is no maximum as far as our memory doesn't run out.
+// As far as I learned, buffer (bytes.buffer) can grow magically and always is the same buffer.
+// But OurByteBuffer created as a slice bytes and can grow but always not the same slice bytes.
+
+// Are there any important performance characteristics? e.g. is it much faster or slower to Write one large amount of data to it than write the same amount of data one byte at a time?
+// yes, for buffer (bytes.buffer) is always faster because working with block of data.
+// for OurByteBuffer which is a slice, it is better to append a large amount of data instead of one byte at a time to trigger the copy process less or create a slice with estimated size we need.
+
+func (b *OurByteBuffer) OurWriteString(text string) {
+	b.myBuff = append(b.myBuff, []byte(text)...)
+}
+
+func (b *OurByteBuffer) OurBytes() []byte {
+	return b.myBuff
+}
+
+// Are there any important memory characteristics? e.g. does an OurByteBuffer always retain all data that was stored in it, or does it free some of its memory after it's been read?
+// when we read from buffer/OurByteBuffer, the pointer move on and we don't access to data read so later garbage collector will frees them as there is no reference to them.
+func (b *OurByteBuffer) OurRead(s []byte) {
```

Comment:
Can you make this actually implement `io.Writer` and `io.Reader`, rathe than having custom `Our*` methods?